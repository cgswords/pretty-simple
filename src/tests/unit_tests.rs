// Copyright 2025 Cameron Swords
// SPDX-License-Identifier: Apache-2.0

use insta::assert_snapshot;

use crate::*;

#[test]
fn column() {
    let doc = Doc::text("prefix").concat_space(Doc::column(|l| Doc::text("| <- column").concat_space(Doc::text(format!("{l}")))));
    let doc = Doc::vsep(vec![0,4,8].into_iter().map(|n| Doc::indent(doc.clone(), n)));
    assert_snapshot!(doc.render(20))
}

#[test]
fn nesting() {
    let doc = Doc::text("prefix").concat_space(Doc::nesting(|l| Doc::brackets(Doc::text("Nested:").concat_space(Doc::text(format!("{l}"))))));
    let doc = Doc::vsep(vec![0,4,8].into_iter().map(|n| Doc::indent(doc.clone(), n)));
    assert_snapshot!(doc.render(20))
}
