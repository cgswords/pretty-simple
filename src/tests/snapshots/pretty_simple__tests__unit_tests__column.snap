---
source: src/tests/unit_tests.rs
expression: doc.render(20)
---
prefix | <- column 7
    prefix | <- column 11
        prefix | <- column 15
